src/script/lua_api/l_vmanip.cpp

```diff
@@ -30,6 +30,91 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "mapgen/mapgen.h"
 #include "voxelalgorithms.h"
 
+#if USE_LUAJIT
+
+extern "C" int VoxelManip_get_volume(void **lvmp)
+{
+	return (*(LuaVoxelManip **)lvmp)->vm->m_area.getVolume();
+}
+
+extern "C" void VoxelManip_get_data(void **lvmp, u16 *data)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	MMVManip *vm = (*(LuaVoxelManip **)lvmp)->vm;
```

Comment:
also try and see if storing a list of valid vm pointers would have a major performance impact and if not use that to verify we didn't get passed an invalid pointer.

Reply:
The null pointer checks are definitely needed, mainly because it's easy to hit e.g. by mistyping the variable name, and a Segmentation Fault error usually lacks information to locate the problem.

But Is it worth the effort to maintain a list and check if the vm is in it? There are just a small bunch of types of variables that can go there. Userdata is one of the few. In other cases, a normal Lua error is raised. It looks overkill to me.